Running-Mastodon/Serving_a_different_domain.md

```diff
@@ -0,0 +1,85 @@
+
+## Webfinger
+
+As said earlier, each Mastodon user has an `acct:` URI, which is used for discovery and authentication.
+When you add `@user@example.org`, a webfinger query is performed. This is done in two steps:
+1. Querying `https://example.org/.well-known/host-meta` (where the domain of the URL matches the domain part of the `acct:` URI)
+   to get information on how to perform the query.
+   This file will indeed contain a URL template of the form `https://somedomain.org/.well-known/webfinger?resource={uri}` that
+   will be used in the second step.
+2. Fill the returned template with the `acct:` URI to be queried and perform the query:
+   `https://somedomain.org/.well-known/webfinger?resource=acct:user@example.org`
+
+## Configuring your Mastodon instance
+
+*DO NOT ATTEMPT TO CONFIGURE YOUR INSTANCE THIS WAY IF YOU DID NOT UNDERSTAND THE ABOVE*
+
+Also, make sure to check the known issues below.
+
+### Configuring Mastodon itself
+
+Mastodon has a two configuration variables in `.env.production` to enable using different domains for your users and Mastodon itself.
+Those two variables are `LOCAL_DOMAIN` and `WEB_DOMAIN`. When the latter is not set, it defaults to the value of
+`LOCAL_DOMAIN`.
+
+*Be extra careful when configuring your Mastodon instance, as changing `LOCAL_DOMAIN` without changing `WEB_DOMAIN`
+may cause remote instances to register different accounts with the same author/actor URI, which will result in federation issues!*
+
+- `LOCAL_DOMAIN` is the domain for which your Mastodon instance has authority, it is the domain used in the `acct:` URI.
+  In our example, `LOCAL_DOMAIN` would be set to `example.org`.
+- `WEB_DOMAIN` is the domain used for any URL generated for your instance, including the author/actor URIs.
+  In our case, that would be `social.example.org`.
+
+### Configuring domain.org
+
+Now, you have Mastodon running at `https://social.example.org` as well as a website at `https://example.org`.
+If you recall how webfinger queries work, the first step is to query `https://example.org/.well-known/host-meta`,
+which will contain an URL template.
+
+Therefore, the easiest way to configure domain.org is to redirect `/.well-known/host-meta` to social.example.org.
+
+With nginx, it would be as simple as adding:
+
+```nginx
+location /.well-known/host-meta {
+       return 301 https://social.example.org$request_uri;
+}
+```
+
+in example.org's server block.
+
+## Known issues
+
+There are a few known issues with Mastodon:
+
+- Current mastodon instances won't correctly process inbound salmon requests from instances where `WEB_DOMAIN`!= `LOCAL_DOMAIN`,
+  and will try resolving `acct:user@WEB_DOMAIN` instead of `acct:user@LOCAL_DOMAIN`.
+  Fortunately, there is a workaround in the master branch to allow your instance to reply to such queries with informations on your actual account.
```

Comment:
Current mastodon instances still don't treat this exactly as they should, but the workaround is indeed in a tagged release, now (since v1.3.0). I should update that.